services/user_service/database_connector.py

```diff
@@ -0,0 +1,64 @@
+import json
+import firebase_admin
+from firebase_admin import credentials
+from firebase_admin import db
+
+#implements CRUD operations interface
+class DatabaseAdmin:
+
+    def __init__(self,databaseURL, credentials_path):
+        cred = credentials.Certificate(credentials_path)
+        firebase_admin.initialize_app(cred,
+        {
+            'databaseURL' : databaseURL,
+        })
+        self.__users_ref = db.reference("users")
+
+
+    def get_all_users(self):
+        return self.__users_ref.get()
+    
+
+    def get_user_by_email(self, email):
+        for user_id in self.get_user_ids():
+            if email == self.__users_ref.child(user_id).child("email").get():
+                return self.__users_ref.child(user_id).get()
```

Comment:
How about error handling? Won't init throw an exception if we have no internet access? How you will know if user was really added?

Reply:
https://github.com/ozgur/python-firebase/issues/85#issuecomment-412225426

Reply:
All right, offline capabilities don't work
Now you can answer three my questions above

Reply:
I will add error handling in the refactoring commit. Basically, I should catch Value Error and FirebaseError (stands for all errors related to the communication with the Firebase remote server)